src/main/kotlin/org/jitsi/nlj/stats/EndpointConnectionStats.kt

```diff
@@ -85,22 +90,23 @@ class EndpointConnectionStats(
         when (packet) {
             is RtcpSrPacket -> {
                 logger.cdebug { "Tracking sent SR packet with compacted timestamp ${packet.senderInfo.compactedNtpTimestamp}" }
-                srSentTimes[Pair(packet.senderSsrc, packet.senderInfo.compactedNtpTimestamp)] =
-                        System.currentTimeMillis()
+                srSentTimes[Pair(packet.senderSsrc, packet.senderInfo.compactedNtpTimestamp)] = clock.instant()
             }
         }
     }
 
-    private fun processReportBlock(receivedTime: Long, reportBlock: RtcpReportBlock) {
+    private fun processReportBlock(receivedTime: Instant, reportBlock: RtcpReportBlock) {
         if (reportBlock.lastSrTimestamp > 0 && reportBlock.delaySinceLastSr > 0) {
             // We need to know when we sent the last SR
-            val srSentTime = srSentTimes.getOrDefault(SsrcAndTimestamp(reportBlock.ssrc, reportBlock.lastSrTimestamp), -1)
-            if (srSentTime > 0) {
+            srSentTimes[SsrcAndTimestamp(reportBlock.ssrc, reportBlock.lastSrTimestamp)]?.let { srSentTime ->
                 // The delaySinceLastSr value is given in 1/65536ths of a second, so divide it by 65.536 to get it
                 // in milliseconds
                 val remoteProcessingDelayMs = reportBlock.delaySinceLastSr / 65.536
-                rtt = receivedTime - srSentTime - remoteProcessingDelayMs
+                rtt = receivedTime.toEpochMilli() - srSentTime.toEpochMilli() - remoteProcessingDelayMs
                 endpointConnectionStatsListeners.forEach { it.onRttUpdate(rtt) }
```

Comment:
Do you want to do math with `Duration` here?  This doesn't make a difference while you're calculating the times from epoch millis to start with, but it'll get you the additional precision for free once we have higher-precision instants.

`remoteProcessingDelay`[Ms] should probably also be a `Duration`

Reply:
for some reason i thought i'd have to do `Duration.between(...)` each time which would've been awkward.  will fix.

Reply:
actually, this is a bit tricky as i've got 2 `Instant`s (`receivedTime` and `srSentTime`) and one `Duration` (`remoteProcessingDelay`)

Reply:
I think you can subtract `Duration`s, so use `Duration.between` then subtract `remoteProcessingDelay`?

Reply:
Yup, just thought of this in the car :) will do that

Reply:
made this change, but have to do some juggling of types (between longs and doubles). i feel like maybe it could be better, but let me know what you think.

Reply:
and it does look like we lose some precision there

Reply:
basically, in order to turn `remoteProcessingDelay` into a `Duration`, i have to convert the value from a double to a Long

Reply:
You should be able to convert it to Long nanoseconds, no? That won't lose any meaningful precision.

Reply:
i was referring to the precision lost in the cast from `Double` to `Long`, before converting to the `Duration`